### ISSUE TITLE
Gas meter definition & fee handling

### ISSUE THREAD

**Issue by eckelj** [planetmint/planetmint-go#3]
Define and test the gas meter definition:

* How are the gas fees defined (minimum, pro node) & verified
* How are the fees collected?
* What happens to the fees? To whom (accounts) are they distributed?

Requirements:
* it should be possible to define a distribution array like : (address1, %),  (address2, %), ... with (cosmos calls this entitlement)

EDIT (22.8.23):
Tasks:
- [x] test GasKV consumption on read and write
- [x] test gas consumption Gas Meter for protobuf byte size consumption 
- [x] test how txs are handled that exceed the BlockGasMeter

**Comment by eckelj** [comment]
We try to charge per byte of the raw protobuf bytes.

**Comment by LaurentMontBlanc** [comment]
TL;DR: Fees are deducted for Gas is currently consumed for the following:
- Tx size
- Signature validation
- Reads/writes from stores


## Findings
By default the AnteHandler is already consuming gas for the size of a transaction with the `ConsumeTxSizeGasDecorator`. The cost per byte is defined in the `AccountKeeper` params. 

Additional gas is consumed for Signature validation based on the type of signature.
